src/crimson/net/ProtocolV2.cc

```diff
@@ -855,16 +858,30 @@ void ProtocolV2::execute_connecting()
             logger().debug("{} UPDATE: gs={}, cc={} for connect",
                            conn, global_seq, client_cookie);
           }
+
+          return wait_write_exit();
+        }).then([this] {
+          if (unlikely(state != state_t::CONNECTING)) {
```

Comment:
> `if (likely(state == state_t::CONNECTING)) {`

IMHO this will complicate the logic because `state != state_t::CONNECTING` is by nature a branched condition which ends up with `abort_protocol()`. And the next continuation expects a return value of `SocketFRef sock`.

Reply:
sorry, i misread the code.